src/wasm-interpreter.h

```diff
@@ -1919,6 +1919,35 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     }
     return Literal(int32_t(data->values.size()));
   }
+  Flow visitStringConcat(StringConcat* curr) { 
+    NOTE_ENTER("StringConcat");
+    Flow flow = visit(curr->left);
+    if (flow.breaking()) {
+      return flow;
+    }
+    auto left = flow.getSingleValue();
+    flow = visit(curr->right);
+    if (flow.breaking()) {
+      return flow;
+    }
+    auto right = flow.getSingleValue();
+    NOTE_EVAL2(left, right);
+    auto leftData = left.getGCData();
+    auto rightData = right.getGCData();
+    if (!leftData || !rightData) {
+      trap("null ref");
+    }
+
+    char array[leftData->values.size() + rightData->values.size()];
+    int i =0;
+    for (Literal l : leftData->values) {
+      array[i++] = l.getInteger();
+    }
+    for (Literal l : rightData->values) {
+      array[i++] = l.getInteger();
+    }
+    return Literal(std::string(array));
```

Comment:
Apologies, I thought I commented this before but maybe I didn't click the right button on that particular comment: this part of the code can be replaced with code like this from StringNew:

https://github.com/WebAssembly/binaryen/blob/880b3676af52a979ed45b6f3ff9a20b853d0c6ea/src/wasm-interpreter.h#L1888-L1895

That is, it is better to use a Literals as the data storage, and then let makeGCData do the rest. (This avoids any questions of whether `std::string` cares about null-termination etc.)